pkg/testsuite/depositcalculator/depositcalculator.go

```diff
@@ -0,0 +1,334 @@
+func WithImmutableMetadataLength(length int) options.Option[Options] {
+	return func(opts *Options) {
+		opts.ImmutableMetadataLength = length
+	}
+}
+
+func getUnlockConditions[T iotago.UnlockCondition](outputType iotago.OutputType, opts *Options) iotago.UnlockConditions[T] {
+	unlockConditions := make(iotago.UnlockConditions[T], 0)
+
+	// Mandatory address unlocks
+	switch outputType {
+	case iotago.OutputBasic, iotago.OutputAccount, iotago.OutputNFT, iotago.OutputDelegation:
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.AddressUnlockCondition{Address: opts.Address}).(T))
+
+	case iotago.OutputAnchor:
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.StateControllerAddressUnlockCondition{Address: opts.StateControllerAddress}).(T))
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.GovernorAddressUnlockCondition{Address: opts.GovernorAddress}).(T))
+
+	case iotago.OutputFoundry:
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.ImmutableAccountUnlockCondition{Address: &iotago.AccountAddress{}}).(T))
+	}
+
+	if opts.StorageDepositReturnAddress != nil {
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.StorageDepositReturnUnlockCondition{ReturnAddress: opts.StorageDepositReturnAddress}).(T))
+	}
+
+	if opts.HasTimelock {
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.TimelockUnlockCondition{}).(T))
+	}
+
+	if opts.ExpirationAddress != nil {
+		unlockConditions = append(unlockConditions, iotago.UnlockCondition(&iotago.ExpirationUnlockCondition{ReturnAddress: opts.ExpirationAddress}).(T))
+	}
+
+	return unlockConditions
+}
+
+func getFeatures[T iotago.Feature](opts *Options) iotago.Features[T] {
+	features := iotago.Features[T]{}
+
+	if opts.SenderAddress != nil {
+		features = append(features, &iotago.SenderFeature{Address: opts.SenderAddress})
+	}
+
+	if opts.IssuerAddress != nil {
+		features = append(features, &iotago.IssuerFeature{Address: opts.IssuerAddress})
+	}
+
+	if opts.MetadataLength > 0 {
+		features = append(features, &iotago.MetadataFeature{
+			Entries: iotago.MetadataFeatureEntries{
+				// configured length - map length prefix - key length prefix - value length prefix
+				"": make([]byte, opts.MetadataLength-int(serix.LengthPrefixTypeAsByte)+int(serix.LengthPrefixTypeAsByte)+int(serix.LengthPrefixTypeAsUint16)),
+			},
+		})
+	}
+
+	if opts.StateMetadataLength > 0 {
+		features = append(features, &iotago.StateMetadataFeature{
+			Entries: iotago.StateMetadataFeatureEntries{
+				// configured length - map length prefix - key length prefix - value length prefix
+				"": make([]byte, opts.StateMetadataLength-int(serix.LengthPrefixTypeAsByte)+int(serix.LengthPrefixTypeAsByte)+int(serix.LengthPrefixTypeAsUint16)),
+			},
+		})
+	}
+
+	if opts.TagLength > 0 {
+		features = append(features, &iotago.TagFeature{Tag: make([]byte, opts.TagLength)})
+	}
```

Comment:
For the metdata, the length we pass in is what the entire metadata feature ends up having in size. For the Tag Feature we'd have length + length prefix which is a bit inconsistent.

So I'd suggest to subtract the length prefix byte here.

If you do that, also update the documentation on `WithTagLength`.

Reply:
I renamed the other funcs to clarify the difference.